src/ChannelServer/World/Monster.cs

```diff
@@ -189,7 +191,18 @@ public void Kill(Character killer)
 			this.DisappearTime = DateTime.Now.AddSeconds(2);
 			killer.GiveExp(exp, classExp, this);
 
+			if (this.spawnZone != null)
+			{
+				this.spawnZone.NotifyEntityDeath(this.Handle);
```

Comment:
This will certainly work, but if I may make a suggestion, what do you think about using events? If Monster had a `Died` event that is raised here, the spawner could subscribe to that when the monster is spawned. That's much more flexible, can be used by any number of other methods and classes, and the Monster doesn't need a reference to its spawn zone, reducing dependencies. The zone simply gets notified about the death and spawns another monster.

Reply:
Interface maybe?

Reply:
What are you proposing exactly?